netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -287,11 +279,22 @@ static ProtocolNegotiator createProtocolNegotiator(
     }
   }
 
+  interface AuthorityChecker {
```

Comment:
Its role could be made more clear if it's named `AuthorityCheckOverride`, and its setters `overrideAuthorityCheck()`.